sub/sd_ass.c

```diff
@@ -294,8 +294,19 @@ static bool is_animated(char *s)
     bool valid_event = false;
     bool valid_tag = false;
     while (*s) {
-        if (!in_tag && s[0] == '{')
+        if (!in_tag) {
+            s = strchr(s, '{');
+            if (!s)
+                break;
             in_tag = true;
+            s++;
+            continue;
+        } else if (in_tag && valid_event) {
+            s = strchr(s, '}');
+            if (!s)
+                break;
+        }
```

Comment:
Shouldn't this be handled below where the existing check for this condition is?

Reply:
It seemed reasonable to do all the big strchr skips up top.